### PR TITLE
Frontend: Callable references to declarations with context parameters KT-73145

### PR DESCRIPTION
**Pull request by cypressious**
After yesterday's outage, the new commits didn't appear in #6088 so I've created a new PR.

### CODE REVIEW

**Comment by ghost** [comment]
### Code Owners

<table><tr><th>Rule</th><th>Owners</th><th>Approval</th></tr><tr><td><code>*Generated.​java</code></td><td></td><td align="center">✅<br><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a> 🔒, <a href="https://github.com/serras"><b><code>@serras</code></b></a>, <a href="https://github.com/dimonchik0036"><b><code>@dimonchik0036</code></b></a> 🔒, <a href="https://github.com/ShikaSD"><b><code>@ShikaSD</code></b></a></td></tr><tr><td><code>/​analysis/​</code></td><td><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-analysis-api">kotlin-analysis-api</a></summary><ul><li><a href="https://github.com/yanex"><b><code>@yanex</code></b></a></li><li><a href="https://github.com/marcopennekamp"><b><code>@marcopennekamp</code></b></a></li><li><a href="https://github.com/abelkov"><b><code>@abelkov</code></b></a></li><li><a href="https://github.com/dimonchik0036"><b><code>@dimonchik0036</code></b></a></li><li><a href="https://github.com/BlaBlaHuman"><b><code>@BlaBlaHuman</code></b></a></li></ul></details></td><td align="center">✅<br><a href="https://github.com/dimonchik0036"><b><code>@dimonchik0036</code></b></a> 🔒</td></tr><tr><td><code>/​compiler/​fir/​</code>, <code>/​compiler/​testData/​diagnostics/​</code></td><td><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-frontend">kotlin-frontend</a></summary><ul><li><a href="https://github.com/ligee"><b><code>@ligee</code></b></a></li><li><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a></li><li><a href="https://github.com/bnorm"><b><code>@bnorm</code></b></a></li><li><a href="https://github.com/mglukhikh"><b><code>@mglukhikh</code></b></a></li><li><a href="https://github.com/KvanTTT"><b><code>@KvanTTT</code></b></a></li><li><a href="https://github.com/cypressious"><b><code>@cypressious</code></b></a></li><li><a href="https://github.com/demiurg906"><b><code>@demiurg906</code></b></a></li><li><a href="https://github.com/FenstonSingel"><b><code>@FenstonSingel</code></b></a></li><li><a href="https://github.com/lunakoly"><b><code>@lunakoly</code></b></a></li><li><a href="https://github.com/agamzikova"><b><code>@agamzikova</code></b></a></li><li><a href="https://github.com/GrigoriiSolnyshkin"><b><code>@GrigoriiSolnyshkin</code></b></a></li><li><a href="https://github.com/Aleksandra-Arsenteva"><b><code>@Aleksandra-Arsenteva</code></b></a></li></ul></details></td><td align="center">✅<br><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a> 🔒</td></tr><tr><td><code>/​compiler/​ir/​ir.​actualization/​</code></td><td><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-common-backend">kotlin-common-backend</a></summary><ul><li><a href="https://github.com/vsukharev"><b><code>@vsukharev</code></b></a></li><li><a href="https://github.com/kunyavskiy"><b><code>@kunyavskiy</code></b></a></li><li><a href="https://github.com/ivandev0"><b><code>@ivandev0</code></b></a></li><li><a href="https://github.com/AnzhelaSukhanova"><b><code>@AnzhelaSukhanova</code></b></a></li><li><a href="https://github.com/mcpiroman"><b><code>@mcpiroman</code></b></a></li><li><a href="https://github.com/ddolovov"><b><code>@ddolovov</code></b></a></li><li><a href="https://github.com/mpiekutowski"><b><code>@mpiekutowski</code></b></a></li></ul></details><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-frontend">kotlin-frontend</a></summary><ul><li><a href="https://github.com/ligee"><b><code>@ligee</code></b></a></li><li><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a></li><li><a href="https://github.com/bnorm"><b><code>@bnorm</code></b></a></li><li><a href="https://github.com/mglukhikh"><b><code>@mglukhikh</code></b></a></li><li><a href="https://github.com/KvanTTT"><b><code>@KvanTTT</code></b></a></li><li><a href="https://github.com/cypressious"><b><code>@cypressious</code></b></a></li><li><a href="https://github.com/demiurg906"><b><code>@demiurg906</code></b></a></li><li><a href="https://github.com/FenstonSingel"><b><code>@FenstonSingel</code></b></a></li><li><a href="https://github.com/lunakoly"><b><code>@lunakoly</code></b></a></li><li><a href="https://github.com/agamzikova"><b><code>@agamzikova</code></b></a></li><li><a href="https://github.com/GrigoriiSolnyshkin"><b><code>@GrigoriiSolnyshkin</code></b></a></li><li><a href="https://github.com/Aleksandra-Arsenteva"><b><code>@Aleksandra-Arsenteva</code></b></a></li></ul></details></td><td align="center">✅<br><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a> 🔒</td></tr><tr><td><code>/​compiler/​testData/​codegen/​box/​</code>, <code>/​compiler/​testData/​ir/​irText/​</code></td><td><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-compiler">kotlin-compiler</a></summary><ul><li><a href="https://github.com/erokhins"><b><code>@erokhins</code></b></a></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-frontend">kotlin-frontend</a></summary><ul><li><a href="https://github.com/ligee"><b><code>@ligee</code></b></a></li><li><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a></li><li><a href="https://github.com/bnorm"><b><code>@bnorm</code></b></a></li><li><a href="https://github.com/mglukhikh"><b><code>@mglukhikh</code></b></a></li><li><a href="https://github.com/KvanTTT"><b><code>@KvanTTT</code></b></a></li><li><a href="https://github.com/cypressious"><b><code>@cypressious</code></b></a></li><li><a href="https://github.com/demiurg906"><b><code>@demiurg906</code></b></a></li><li><a href="https://github.com/FenstonSingel"><b><code>@FenstonSingel</code></b></a></li><li><a href="https://github.com/lunakoly"><b><code>@lunakoly</code></b></a></li><li><a href="https://github.com/agamzikova"><b><code>@agamzikova</code></b></a></li><li><a href="https://github.com/GrigoriiSolnyshkin"><b><code>@GrigoriiSolnyshkin</code></b></a></li><li><a href="https://github.com/Aleksandra-Arsenteva"><b><code>@Aleksandra-Arsenteva</code></b></a></li></ul></details></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-wasm">kotlin-wasm</a></summary><ul><li><a href="https://github.com/bashor"><b><code>@bashor</code></b></a></li><li><a href="https://github.com/karlosz"><b><code>@karlosz</code></b></a></li><li><a href="https://github.com/ilgonmic"><b><code>@ilgonmic</code></b></a></li><li><a href="https://github.com/jmrtsh"><b><code>@jmrtsh</code></b></a></li><li><a href="https://github.com/igoriakovlev"><b><code>@igoriakovlev</code></b></a></li><li><a href="https://github.com/agamzikova"><b><code>@agamzikova</code></b></a></li><li><a href="https://github.com/alex28sh"><b><code>@alex28sh</code></b></a></li><li><a href="https://github.com/ozzush"><b><code>@ozzush</code></b></a></li></ul></details></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-jvm">kotlin-jvm</a></summary><ul><li><a href="https://github.com/udalov"><b><code>@udalov</code></b></a></li><li><a href="https://github.com/ilmirus"><b><code>@ilmirus</code></b></a></li><li><a href="https://github.com/grechkovlad"><b><code>@grechkovlad</code></b></a></li><li><a href="https://github.com/MaxVerevkin"><b><code>@MaxVerevkin</code></b></a></li><li><a href="https://github.com/zhelenskiy"><b><code>@zhelenskiy</code></b></a></li><li><a href="https://github.com/aShalygin"><b><code>@aShalygin</code></b></a></li><li><a href="https://github.com/andrey-mogilev"><b><code>@andrey-mogilev</code></b></a></li></ul></details></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-js">kotlin-js</a></summary><ul><li><a href="https://github.com/JSMonk"><b><code>@JSMonk</code></b></a></li><li><a href="https://github.com/broadwaylamb"><b><code>@broadwaylamb</code></b></a></li><li><a href="https://github.com/seclerp"><b><code>@seclerp</code></b></a></li><li><a href="https://github.com/AnzhelaSukhanova"><b><code>@AnzhelaSukhanova</code></b></a></li></ul></details></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-common-backend">kotlin-common-backend</a></summary><ul><li><a href="https://github.com/vsukharev"><b><code>@vsukharev</code></b></a></li><li><a href="https://github.com/kunyavskiy"><b><code>@kunyavskiy</code></b></a></li><li><a href="https://github.com/ivandev0"><b><code>@ivandev0</code></b></a></li><li><a href="https://github.com/AnzhelaSukhanova"><b><code>@AnzhelaSukhanova</code></b></a></li><li><a href="https://github.com/mcpiroman"><b><code>@mcpiroman</code></b></a></li><li><a href="https://github.com/ddolovov"><b><code>@ddolovov</code></b></a></li><li><a href="https://github.com/mpiekutowski"><b><code>@mpiekutowski</code></b></a></li></ul></details></li><li><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-native">kotlin-native</a></summary><ul><li><a href="https://github.com/mMaxy"><b><code>@mMaxy</code></b></a></li><li><a href="https://github.com/projedi"><b><code>@projedi</code></b></a></li><li><a href="https://github.com/anton-bannykh"><b><code>@anton-bannykh</code></b></a></li><li><a href="https://github.com/sbogolepov"><b><code>@sbogolepov</code></b></a></li><li><a href="https://github.com/DmT021"><b><code>@DmT021</code></b></a></li><li><a href="https://github.com/haitaka"><b><code>@haitaka</code></b></a></li><li><a href="https://github.com/glukianets"><b><code>@glukianets</code></b></a></li><li><a href="https://github.com/azakharenko"><b><code>@azakharenko</code></b></a></li><li><a href="https://github.com/homuroll"><b><code>@homuroll</code></b></a></li><li><a href="https://github.com/SvyatoslavScherbina"><b><code>@SvyatoslavScherbina</code></b></a></li></ul></details></li></ul></details></td><td align="center">✅<br><a href="https://github.com/dzharkov"><b><code>@dzharkov</code></b></a> 🔒</td></tr><tr><td><code>/​core/​language.​version-​settings/​src/​org/​jetbrains/​kotlin/​config/​LanguageVersionSettings.​kt</code></td><td><details><summary><a href="https://github.com/orgs/JetBrains/teams/kotlin-language-evolution">kotlin-language-evolution</a></summary><ul><li><a href="https://github.com/serras"><b><code>@serras</code></b></a></li><li><a href="https://github.com/jesyspa"><b><code>@jesyspa</code></b></a></li><li><a href="https://github.com/ice-phoenix"><b><code>@ice-phoenix</code></b></a></li><li><a href="https://github.com/zarechenskiy"><b><code>@zarechenskiy</code></b></a></li><li><a href="https://github.com/watc4d0gg"><b><code>@watc4d0gg</code></b></a></li><li><a href="https://github.com/nikitabobko"><b><code>@nikitabobko</code></b></a></li><li><a href="https://github.com/FenstonSingel"><b><code>@FenstonSingel</code></b></a></li><li><a href="https://github.com/InversionSpaces"><b><code>@InversionSpaces</code></b></a></li><li><a href="https://github.com/e2e4b6b7"><b><code>@e2e4b6b7</code></b></a></li><li><a href="https://github.com/YuliaKorolenko"><b><code>@YuliaKorolenko</code></b></a></li></ul></details></td><td align="center">✅<br><a href="https://github.com/serras"><b><code>@serras</code></b></a></td></tr></table>

File: compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AdapterGenerator.kt [hunk 1]
```diff
@@ -203,32 +300,45 @@ class AdapterGenerator(
...
+                boundReceiver?.let {
                     if (boundDispatchReceiver != null && boundExtensionReceiver != null) {
                         error("Bound callable references can't have both receivers: ${callableReferenceAccess.render()}")
                     } else {
                         this += createAdapterParameter(
                             irAdapterFunction,
                             Name.identifier("receiver"),
-                            boundReceiver.type,
+                            it.type,
                             IrDeclarationOrigin.ADAPTER_PARAMETER_FOR_CALLABLE_REFERENCE,
```
Comment (kunyavskiy) [inline compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AdapterGenerator.kt:342]: As I see in UpgradeCallableReference lowering, in case of bound receiver there is a special origin, which if I remember correctly was important for some debug info generation. 
It's better to ask someone from jvm team, or just restore keep it as it was 

You can check IrFunction.flattenParameters in UpgradeCallableReferenceLowering for details. It should be visible on some bytecode listing tests, if you enable it for everything.

Comment (cypressious) [reply compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AdapterGenerator.kt:342]: I only see one failing bytecode test `compiler/testData/codegen/bytecodeListing/sam/indySamConversionViaProxyFun.kt` but it doesn't seem to be caused by parameter origins but rather by the `isInline` flag of the `invokeFunction`.

Comment (cypressious) [reply compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AdapterGenerator.kt:342]: Also, it looks like the logic in `flattenParameters` is broken by the commit `[FIR2IR] Generate bound extension receiver parameter of adapter function as regular parameter`. I'm doing a dry-run to understand if this breaks any tests.

**Review by dzharkov — APPROVED** [review]
/final

**Review by serras — APPROVED** [review]

**Review by dimonchik0036 — DISMISSED** [review]
Could you please apply the patch as you did for `testData/diagnostics/tests/contextParameters/callableReference.kt`?
```patch
Index: analysis/analysis-api/testData/components/resolver/allByPsi/withErrors/callableReferenceWithErrors.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/analysis/analysis-api/testData/components/resolver/allByPsi/withErrors/callableReferenceWithErrors.kt b/analysis/analysis-api/testData/components/resolver/allByPsi/withErrors/callableReferenceWithErrors.kt
--- a/analysis/analysis-api/testData/components/resolver/allByPsi/withErrors/callableReferenceWithErrors.kt	(revision 6dbbccca686497cb4509eba33e8ce93f6bf9a80d)
+++ b/analysis/analysis-api/testData/components/resolver/allByPsi/withErrors/callableReferenceWithErrors.kt	(date 1780044544159)
@@ -76,4 +76,12 @@
     ::withContextProperty(LoggerContext())
 }
 
+context(c: LoggerContext)
+fun callableReferenceWithContextArgumentsWithContext() {
+    ::withContext
+    ::withContext(LoggerContext(), 1)
+    ::withContextProperty
+    ::withContextProperty(LoggerContext())
+}
+
 // LANGUAGE: +ContextParameters +ExplicitContextArguments
Index: analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext.kt b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext.kt
new file mode 100644
--- /dev/null	(date 1780044772137)
+++ b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext.kt	(date 1780044772137)
@@ -0,0 +1,10 @@
+context(param1: Int, param2: String)
+val foo: Boolean get() = false
+
+context(param1: Int, param2: String)
+fun usage() {
+    <expr>::foo</expr>
+}
+
+// LANGUAGE: +ContextParameters
+// COMPILATION_ERRORS
Index: analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext_rhs.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext_rhs.kt b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext_rhs.kt
new file mode 100644
--- /dev/null	(date 1780044772130)
+++ b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnPropertyWithContextParametersWithContext_rhs.kt	(date 1780044772130)
@@ -0,0 +1,10 @@
+context(param1: Int, param2: String)
+val foo: Boolean get() = false
+
+context(param1: Int, param2: String)
+fun usage() {
+    ::<expr>foo</expr>
+}
+
+// LANGUAGE: +ContextParameters
+// COMPILATION_ERRORS
Index: analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext.kt b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext.kt
new file mode 100644
--- /dev/null	(date 1780044772140)
+++ b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext.kt	(date 1780044772140)
@@ -0,0 +1,12 @@
+context(param1: Int, param2: String)
+fun foo(b: Boolean) {
+
+}
+
+context(param1: Int, param2: String)
+fun usage() {
+    <expr>::foo</expr>
+}
+
+// LANGUAGE: +ContextParameters
+// COMPILATION_ERRORS
Index: analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext_rhs.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext_rhs.kt b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext_rhs.kt
new file mode 100644
--- /dev/null	(date 1780044772134)
+++ b/analysis/analysis-api/testData/components/resolver/singleByPsi/contextParameters/callableReferenceOnFunctionWithContextParametersWithContext_rhs.kt	(date 1780044772134)
@@ -0,0 +1,12 @@
+context(param1: Int, param2: String)
+fun foo(b: Boolean) {
+
+}
+
+context(param1: Int, param2: String)
+fun usage() {
+    ::<expr>foo</expr>
+}
+
+// LANGUAGE: +ContextParameters
+// COMPILATION_ERRORS
```

**Comment by cypressious** [comment]
@dimonchik0036 `/final` needs to go into the review comment, not a regular comment :(

**Review by dimonchik0036 — APPROVED** [review]
/final

**Comment by dimonchik0036** [comment]
> @dimonchik0036 `/final` needs to go into the review comment, not a regular comment :(

That's quite unfortunate because then I cannot leave the real feedback in the review comment, so I'm forced to either attach it to some code or write it separately later

File: plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt [hunk 2]
```diff
@@ -35,10 +35,10 @@ fun Ref(cls: Cls, content: Function2<Composer, Int, Unit>, %composer: Composer?,
       sourceInformationMarkerStart(%composer, <>, "CC(remember):Test.kt#9igjgp")
       val tmp0_group = %composer.cache(%dirty and 0b1110 == 0b0100 || %dirty and 0b1000 != 0 && %composer.changedInstance(tmp0)) {
         @Composable
-        fun Cls.Fn(%composer: Composer?, %changed: Int) {
+        fun Fn(receiver: Cls, %composer: Composer?, %changed: Int) {
           receiver.Fn(0, %composer, Cls.%stable shl 0b0011 or 0b01110000 and %changed shl 0b0011, 0b0001)
         }
-        tmp0::Fn
+        ::Fn
```
Comment (ShikaSD) [inline plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: I am not sure this is right, if I read this correctly it got transformed from adapter reference with a receiver into a regular function, which adds one more parameter to the function reference

Comment (ShikaSD) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: The function expects Function2<Composer, Int>, and this is effectively Function3

Comment (cypressious) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: Yes, looks like it. Do you know which code is responsible so I can try to fix it?

Comment (ShikaSD) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: Should be https://github.com/JetBrains/kotlin/blob/e15717c1fc083b5762b426eb91e10e8cc982f9dc/plugins/compose/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/ComposerParamTransformer.kt#L219

Comment (cypressious) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: Btw, according to https://github.com/JetBrains/kotlin/blob/4bd5ce2adce17a420e0c6e48d8ba57630fb88873/plugins/compose/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/ComposerParamTransformer.kt#L293, it seems Compose will require some work for context-bound callable references.

Comment (ShikaSD) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: Yeah, I was just copying your homework :)
Should be mostly equivalent to the rest of the compiler, the special treatment was only required to handle defaults and small edge cases

Comment (ShikaSD) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: You can file a separate bug and assign to me to finish the Compose part.

Comment (cypressious) [reply plugins/compose/compiler-hosted/integration-tests/src/jvmTest/resources/golden/androidx.compose.compiler.plugins.kotlin.FunctionReferenceTransformTests/reference_dispatch_adapted.txt]: I think it was just a renderering bug. But I reverted the change that caused it anyway.

**Review by ShikaSD — APPROVED** [review]
